### PR TITLE
adding keymap for pistachio

### PR DESCRIPTION
**Pull request by 7-rate**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pistachio/keymaps/rate/keymap.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+extern rgblight_config_t rgblight_config;
```
Comment (drashna) [inline keyboards/pistachio/keymaps/rate/keymap.c:19]:
```suggestion
```

File: keyboards/pistachio/keymaps/rate/keymap.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      RGB_VAI,  _______, _______, _______, _______, _______,  _______, _______,       _______, _______, _______, _______,  KC_PMNS, _______, _______, _______, _______,
+      RGB_VAD,  KC_7,    KC_8,    KC_9, _______, _______,  _______,                   KC_7,    KC_8,    KC_9,    _______,  _______, _______, _______,          _______,
+      _______,  KC_4,    KC_5,    KC_6, _______, _______,  _______,                   KC_4,    KC_5,    KC_6,    _______,  KC_PPLS, KC_PAST, _______, _______, _______,
+      RGB_HUI,  KC_1,    KC_2,    KC_3, KC_0, JP_DOT,  _______,                       KC_1,    KC_2,    KC_3,    _______,  KC_PSLS,          _______, _______, RGB_SAI,
+      RGB_HUD,  _______, _______, _______, _______, _______,  _______,                KC_0,          _______,    JP_DOT,   _______, _______, _______, _______, RGB_SAD
+    )
+};
+
+
+int RGB_current_mode;
```
Comment (drashna) [inline keyboards/pistachio/keymaps/rate/keymap.c:59]:
```suggestion
```

File: keyboards/pistachio/keymaps/rate/keymap.c [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  bool result = false;
+  switch (keycode) {
+      case RGB_RST:
+          if (record->event.pressed) {
+            eeconfig_update_rgblight_default();
+            rgblight_enable();
+            RGB_current_mode = rgblight_config.mode;
```
Comment (drashna) [inline keyboards/pistachio/keymaps/rate/keymap.c:67]:
```suggestion
```

File: keyboards/pistachio/keymaps/rate/keymap.c [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      _______,  KC_4,    KC_5,    KC_6, _______, _______,  _______,                   KC_4,    KC_5,    KC_6,    _______,  KC_PPLS, KC_PAST, _______, _______, _______,
+      RGB_HUI,  KC_1,    KC_2,    KC_3, KC_0, JP_DOT,  _______,                       KC_1,    KC_2,    KC_3,    _______,  KC_PSLS,          _______, _______, RGB_SAI,
+      RGB_HUD,  _______, _______, _______, _______, _______,  _______,                KC_0,          _______,    JP_DOT,   _______, _______, _______, _______, RGB_SAD
+    )
+};
+
+
+int RGB_current_mode;
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  bool result = false;
```
Comment (drashna) [inline keyboards/pistachio/keymaps/rate/keymap.c:61]:
```suggestion
```

File: keyboards/pistachio/keymaps/rate/keymap.c [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      case RGB_RST:
+          if (record->event.pressed) {
+            eeconfig_update_rgblight_default();
+            rgblight_enable();
+            RGB_current_mode = rgblight_config.mode;
+          }
+        break;
+    default:
+      result = true;
+      break;
```
Comment (drashna) [inline keyboards/pistachio/keymaps/rate/keymap.c:72]:
```suggestion
```

File: keyboards/pistachio/keymaps/rate/keymap.c [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  return result;
```
Comment (drashna) [inline keyboards/pistachio/keymaps/rate/keymap.c:75]:
```suggestion
  return true;
```

File: keyboards/pistachio/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+extern rgblight_config_t rgblight_config;
```
Comment (drashna) [inline keyboards/pistachio/keymaps/via/keymap.c:19]:
```suggestion
```

File: keyboards/pistachio/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+            RGB_current_mode = rgblight_config.mode;
+          }
+        break;
+    default:
+      result = true;
+      break;
+  }
+
+  return result;
+}
```
Comment (drashna) [inline keyboards/pistachio/keymaps/via/keymap.c:85]:
```suggestion
bool process_record_user(uint16_t keycode, keyrecord_t *record) {
    switch (keycode) {
        case RGB_RST:
            if (record->event.pressed) {
                eeconfig_update_rgblight_default();
                rgblight_enable();
            }
            break;
    }

    return true;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by 7-rate** [comment]
Thanks for the review.
I have responded to your suggestion.

**Review by drashna — APPROVED** [review]

**Comment by 7-rate** [comment]
What else can I do to merge?

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
> What else can I do to merge?

It can take a while for us to get to PRs, as we do this in our free time.  

And generally, we want two approved reviews before a PR gets merged.

**Review by fauxpark — APPROVED** [review]

**Pull request merged**